lib/chai/core/assertions.js

```diff
@@ -1185,17 +1204,29 @@ module.exports = function (chai, _) {
 
     // Has any
     if (any) {
-      var intersection = expected.filter(function(key) {
-        return ~actual.indexOf(key);
+      ok = expected.some(function(expectedKey) {
+        if (isMapOrSet) {
+          return actual.some(function(actualKey) {
+            return _.eql(expectedKey, actualKey);
```

Comment:
If I'm not mistaken; _.eql will use deep equality here. I'm not sure we want deep equality without the `.deep` qualifier. A `===` might work better here.